llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -1265,10 +1272,15 @@ bool WaitcntGeneratorPreGFX12::applyPreexistingWaitcnt(
   MachineInstr *WaitcntInstr = nullptr;
   MachineInstr *WaitcntVsCntInstr = nullptr;
 
+  LLVM_DEBUG(dbgs() << "PreGFX12::applyPreexistingWaitcnt at: " << *It << "\n");
+
   for (auto &II :
        make_early_inc_range(make_range(OldWaitcntInstr.getIterator(), It))) {
-    if (II.isMetaInstruction())
+    LLVM_DEBUG(dbgs() << "pre-existing iter: " << II << "\n");
```

Comment:
```suggestion
    LLVM_DEBUG(dbgs() << "pre-existing iter: " << II);
```

Reply:
Can also move this to the next print

Reply:
I don't understand. The next print is inside a conditional ...